### PR TITLE
Use thiserror instead of snafu

### PR DESCRIPTION
**Pull request by killercup**
Less magic, more diff

### CODE REVIEW

File: src/cli/from_cldr.rs [hunk 1]
```diff
@@ -50,36 +49,26 @@ pub fn cldr_to_kbdgen(output: &Path, bundle_name: &str) -> Result<(), Error> {
...
-    #[snafu(display("Could write kbdgen bundle"))]
-    CannotSave {
-        source: crate::SaveError,
-        backtrace: snafu::Backtrace,
-    },
+    #[error("Updating CLDR repo failed")]
+    CldrRepoUpdate { source: crate::cli::repos::Error },
+    #[error("No locale selected")]
+    NoLocaleSelected,
+    #[error("Could load CLDR file")]
```
Comment (bbqsrc) [inline src/cli/from_cldr.rs]: Typo

File: src/cli/from_cldr.rs [hunk 2]
```diff
@@ -50,36 +49,26 @@ pub fn cldr_to_kbdgen(output: &Path, bundle_name: &str) -> Result<(), Error> {
...
+    CannotOpenFile { source: std::io::Error },
+    #[error("Could load CLDR file")]
```
Comment (bbqsrc) [inline src/cli/from_cldr.rs]: Typo

File: src/cli/from_cldr.rs [hunk 3]
```diff
@@ -50,36 +49,26 @@ pub fn cldr_to_kbdgen(output: &Path, bundle_name: &str) -> Result<(), Error> {
...
+    CannotReadXml { source: serde_xml_rs::Error },
+    #[error("Could write kbdgen bundle")]
```
Comment (bbqsrc) [inline src/cli/from_cldr.rs]: Typo

File: src/cli/from_xkb.rs [hunk 4]
```diff
@@ -390,50 +408,32 @@ impl AsRef<str> for Codepoint {
...
+    #[error("No locale selected")]
+    NoLocaleSelected,
+    #[error("Could load XKB file `{}`", path.display())]
     CannotOpenFile {
         path: PathBuf,
         source: std::io::Error,
-        backtrace: snafu::Backtrace,
     },
-    #[snafu(display("Could read XKB file `{}`", path.display()))]
+    #[error("Could read XKB file `{}`", path.display())]
```
Comment (bbqsrc) [inline src/cli/from_xkb.rs]: Typo

File: src/cli/repos.rs [hunk 5]
```diff
@@ -23,31 +22,29 @@ pub fn update_repo(name: &str, dir: &Path, repo: &str) -> Result<(), Error> {
...
-        source: std::io::Error,
-        backtrace: snafu::Backtrace,
-    },
-    #[snafu(display("Failed to pull CLDR repo changes"))]
-    RepoUpdateFailed {
-        source: std::io::Error,
-        backtrace: snafu::Backtrace,
-    },
+    #[error("Failed to cloning CLDR repo")]
+    RepoCloneFailed { source: std::io::Error },
```
Comment (bbqsrc) [inline src/cli/repos.rs:47]: Typo

File: src/cli/to_m17n_mim.rs [hunk 6]
```diff
@@ -231,56 +243,31 @@ fn dead_key_transforms(
...
-    InvalidCharacterCodeIndex {
-        source: MimConversion,
-        backtrace: snafu::Backtrace,
-    },
-    #[snafu(display("Cannot create transform map"))]
-    CannotCreateTransformMap {
-        source: MimConversion,
-        backtrace: snafu::Backtrace,
     },
+    #[error("Could transform to MIM")]
```
Comment (bbqsrc) [inline src/cli/to_m17n_mim.rs]: This plus next 5, typo

File: src/cli/to_xkb.rs [hunk 7]
```diff
@@ -76,38 +84,27 @@ pub struct Options {
...
         path: PathBuf,
         source: std::io::Error,
-        backtrace: snafu::Backtrace,
-    },
-    #[snafu(display("Could transform to XKB"))]
-    CannotSerializeXkb {
-        source: std::io::Error,
-        backtrace: snafu::Backtrace,
     },
+    #[error("Could transform to XKB")]
```
Comment (bbqsrc) [inline src/cli/to_xkb.rs]: Typo

File: src/m17n_mim/mod.rs [hunk 8]
```diff
@@ -380,47 +378,39 @@ impl TryFrom<String> for Integer {
...
-        input: String,
-        backtrace: snafu::Backtrace,
-    },
-    #[snafu(display("Could not map index `{}` to a character code", index))]
-    InvalidCharactorCodeIndex {
-        index: usize,
-        backtrace: snafu::Backtrace,
-    },
-    #[snafu(display(
+    #[error("Could serialize MIM symbol")]
```
Comment (bbqsrc) [inline src/m17n_mim/mod.rs]: Typo

File: src/m17n_mim/mod.rs [hunk 9]
```diff
@@ -380,47 +378,39 @@ impl TryFrom<String> for Integer {
...
+    InvalidSymbol,
+    #[error("Could serialize MIM text")]
```
Comment (bbqsrc) [inline src/m17n_mim/mod.rs]: Typo

File: src/m17n_mim/mod.rs [hunk 10]
```diff
@@ -380,47 +378,39 @@ impl TryFrom<String> for Integer {
...
+    InvalidText,
+    #[error("Could serialize MIM key combo: {}", input)]
```
Comment (bbqsrc) [inline src/m17n_mim/mod.rs]: Typo

**Review by bbqsrc — CHANGES_REQUESTED** [review]

**Comment by killercup** [comment]
Haha, those are some great typos

**Pull request merged**